libpromises/evalfunction.c

```diff
@@ -1600,9 +1600,10 @@ static FnCallResult FnCallExecResult(ARG_UNUSED EvalContext *ctx, ARG_UNUSED con
         return FnFailure();
     }
 
-    char buffer[CF_EXPANDSIZE];
+    size_t buffer_size = CF_EXPANDSIZE;
+    char *buffer = xcalloc(1, buffer_size);
```

Comment:
There was a missing ``free()`` as well. Added in https://github.com/cfengine/core/commit/cccbed6d8fcb8b9522d8da3eba7502109220df9a.